src/Chainweb/CutDB.hs

```diff
@@ -297,31 +303,41 @@ processCuts logFun headerStore payloadStore queue cutVar = queueToStream
         )
     & S.effects
   where
+    -- | Broadcast the newest estimated `BlockHeight` of the network to other
+    -- components of this node.
+    --
+    updateNetworkHeight :: CutHashes -> IO ()
+    updateNetworkHeight = atomically . writeTVar networkHeight . Just . cutHashesHeight
```

Comment:
@larskuhtz is there a chance that a `Cut` could come in that would significantly drop the agreed-upon Cut Height?

Reply:
There will be incoming cuts with very low height, but the agreed upon height should never decrease. Cuts of very low height should be filtered out immediately by the cut processor.

Reply:
There is some interference with #161 here: with the changes from that PR, if a cuts that are far ahead, would be dropped, too. So we would see in the processor only cuts that are at most 1000 blocks ahead. Once we get close to those we would start mining, even though the network is much further ahead.

One solution could be to sample the current height after we drop old cuts and before we drop far-ahead cuts.